### PR TITLE
executor: split write.go by executor

### PR DESCRIPTION
**Pull request by jackysp**
The current write.go is so large, split it by executor.
PTAL @coocood @tiancaiamao

### CODE REVIEW

**Comment by ngaut** [comment]
Good job.

File: executor/write.go [hunk 1]
```diff
@@ -181,1277 +173,14 @@ func updateRecord(ctx sessionctx.Context, h int64, oldData, newData []types.Datu
...
-		}
-		e.rowCount = uint64(i)
-		rows[i], err = e.getRow(cols, list, i)
-		if err != nil {
-			return nil, errors.Trace(err)
-		}
-	}
-	return
-}
+const (
```
Comment (shenli) [inline executor/write.go]: This could be merged with in`sert_common.go`.

**Comment by zz-jason** [comment]
LGTM

File: executor/write.go (hunk 1 above)
Comment (jackysp) [reply executor/write.go]: Unfortunately, `DirtyTableAddRow` is used by `write.go` now.

**Pull request merged**